src/utils/reviews/getReviewsAggregateOptions.js

```diff
@@ -0,0 +1,146 @@
+      }
+    },
+    {
+      $lookup: {
+        from: 'offers',
+        let: { offerId: '$offer' },
+        pipeline: [
+          {
+            $match: {
+              $expr: {
+                $and: [{ $ne: [{ $type: '$$offerId' }, 'missing'] }, { $ne: [{ $type: '$$offerId' }, 'null'] }]
+              }
+            }
+          },
+          { $match: { $expr: { $eq: ['$_id', '$$offerId'] } } },
+          {
+            $lookup: {
+              from: 'categories',
+              localField: 'category',
+              foreignField: '_id',
+              as: 'category'
+            }
+          },
+          {
+            $unwind: {
+              path: '$category',
+              preserveNullAndEmptyArrays: true
+            }
+          },
+          {
+            $lookup: {
+              from: 'subjects',
+              localField: 'subject',
+              foreignField: '_id',
+              as: 'subject'
+            }
+          },
+          {
+            $unwind: {
+              path: '$subject',
+              preserveNullAndEmptyArrays: true
+            }
+          }
+        ],
+        as: 'offer'
+      }
+    },
+    {
+      $unwind: {
+        path: '$offer',
+        preserveNullAndEmptyArrays: true
+      }
+    },
+    {
+      $lookup: {
+        from: 'cooperation',
+        let: {
+          offerId: '$offer._id',
+          authorId: '$author._id',
+          targetUserId: '$targetUserId'
+        },
+        pipeline: [
+          {
+            $match: {
+              $expr: {
+                $and: [
+                  { $eq: ['$offer', '$$offerId'] },
+                  {
+                    $or: [
+                      {
+                        $and: [{ $eq: ['$receiver', '$$targetUserId'] }, { $eq: ['$initiator', '$$authorId'] }]
+                      },
+                      {
+                        $and: [{ $eq: ['$receiver', '$$authorId'] }, { $eq: ['$initiator', '$$targetUserId'] }]
+                      }
+                    ]
+                  }
+                ]
+              }
+            }
+          }
+        ],
+        as: 'cooperation'
+      }
+    },
+    {
+      $unwind: {
+        path: '$cooperation',
+        preserveNullAndEmptyArrays: true
+      }
+    },
+    {
+      $project: {
+        'offer.subject.name': 1,
+        'offer.subject._id': 1,
+        'offer.category.name': 1,
+        'offer.category._id': 1,
+        'offer._id': 1,
+        'author._id': 1,
+        'author.firstName': 1,
+        'author.lastName': 1,
+        'author.photo': 1,
+        proficiencyLevel: '$cooperation.proficiencyLevel',
+        comment: 1,
+        rating: 1,
+        targetUserId: 1,
+        targetUserRole: 1,
+        createdAt: 1,
+        updatedAt: 1
+      }
+    }
+  ]
+
+  if (Number.isInteger(skip) && skip > 0) {
+    pipeline.push({ $skip: skip })
+  }
+
+  if (Number.isInteger(limit) && limit > 0) {
+    pipeline.push({ $limit: limit })
+  }
```

Comment:
Maybe it would be better to move `Number.isInteger` checks somethere outside? Probably you can assign 0 as default value (instead of `null` [here](https://github.com/ita-social-projects/SpaceToStudy-BackEnd/blob/6a1170c359f5de76fb82b53c6ba428475eba3c44/src/services/review.js#L12)) and it will fix this isInteger check

Reply:
I've made it like that, because of how the limit query works in mongoose that is used [for count](https://github.com/ita-social-projects/SpaceToStudy-BackEnd/blob/6a1170c359f5de76fb82b53c6ba428475eba3c44/src/services/review.js#L15). If limit is undefined or null, the query skips the limit without any problems. But if limit is set to 0, it throws an error saying, "The limit must be positive."
And the MongoDB aggregation query (used for pipeline), throws an error whether limit is 0, undefined, or null - all of them.

Reply:
Okay, I got it. Sometimes it's a bad sign of a framework (mongoose and mongodb in this example) when you struggle with the framework itself to develop something instead of just using it.